src/plugins/data/common/osd_field_types/osd_field_types.ts

```diff
@@ -84,3 +84,43 @@ export const setOsdFieldOverrides = (newOverrides: { [key: string]: any } | unde
 export const getOsdFieldOverrides = (): { [key: string]: any } => {
   return osdFieldOverrides;
 };
+
+/**
+ *  Mapping function from SQL_TYPES to OSD_FIELD_TYPES
+ *
+ *  @param {OPENSEARCH_SQL_TYPES} sqlType
+ *  @return {OSD_FIELD_TYPES}
+ */
+export function castSQLTypeToOSDFieldType(sqlType: OPENSEARCH_SQL_TYPES): OSD_FIELD_TYPES {
```

Comment:
this is one way did we want to consider making [OsdFieldType](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/common/osd_field_types/osd_field_type.ts#L37) have an optional property like
```typescript
public readonly sqlTypes: readonly OPENSEARCH_SQL_FIELD_TYPES[];
```

Reply:
Updated the implementation for consistency. Now it is similar to the esTypes  [ee287e8](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8337/commits/ee287e8f87a614516ee49c2f7946145f3ef91092)